security/libkmip-tests.py

```diff
@@ -31,16 +31,31 @@ def setUp(self):
         '''
         # Check for basic utilities
         smm = SoftwareManager()
+        detected_distro = distro.detect()
         for package in ['gcc', 'make', 'autoconf']:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/OpenKMIP/libkmip/archive/refs/heads/master.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'libkmip-master')
-        if build.make(self.sourcedir):
-            self.fail("Failed to compile libkmip")
-        os.chdir(self.sourcedir)
+        run_type = self.params.get('type', default='upstream')
```

Comment:
@nasastry make default parameter as distro as discussed

Reply:
I would suggest keep upstream as default... the reason being not all the test machine configure with SRC repo.. so it will fail. but upstream src is always available  @PraveenPenguin what do you suggest ?